ecommerce/extensions/payment/processors/liqpay.py

```diff
@@ -0,0 +1,221 @@
+""" LiqPay payment processing. """
+from __future__ import unicode_literals
+
+import logging
+
+import hashlib
+import json
+import requests
+from base64 import b64decode, b64encode
+from decimal import Decimal
+from django.urls import reverse
+from oscar.apps.payment.exceptions import GatewayError
+from urlparse import urljoin
+
+from ecommerce.core.url_utils import get_ecommerce_url
+from ecommerce.extensions.checkout.utils import get_receipt_page_url
+from ecommerce.extensions.payment.exceptions import DuplicateReferenceNumber, InvalidSignatureError
+from ecommerce.extensions.payment.processors import BasePaymentProcessor, HandledProcessorResponse
+
+
+logger = logging.getLogger(__name__)
+
+
+class Liqpay(BasePaymentProcessor):
+    """
+    Constructs a new instance of the Liqpay processor (October 2019)
+
+    For reference, see https://www.liqpay.ua/documentation/en/api/home/.
+    """
+
+    NAME = 'liqpay'
+
+    def __init__(self, site):
+        """
+        Constructs a new instance of the Liqpay processor.
+
+        Raises:
+            KeyError: If no settings configured for this payment processor.
+        """
+        super(Liqpay, self).__init__(site)
+        configuration = self.configuration
+        self.payment_page_url = configuration['payment_page_url']
+        self.public_key = configuration['public_key']
+        self.private_key = configuration['private_key']
+        self.host = configuration['host']
+        self.sandbox_mode = configuration['sandbox_mode']
+        self.version = configuration['version']
```

Comment:
As common practice, to build class instance from whatever you want (string, json, dict, list) there is a `classmethod` approach. Consider it if think that it makes sense for you and your task.